### PR TITLE
model parametrization

### PR DESCRIPTION
**Pull request by ltetrel**
This PR adds a fully dynamic and parametrized GCN model.
It suports:
* number of layers (both GCN and fc)
* filters definition (size, number, for both GCN and linear)
* activation function
* normalizations (batch norm and GCN)
* dropout and pooling (edge pooling still not working)

### CODE REVIEW

**Comment by ltetrel** [comment]
Sorry for the mess, I f***** up the merge conflict. git can be really really tricky sometimes...

File: gcn_package/models/gcn.py [hunk 1]
```diff
@@ -48,41 +48,132 @@ def forward(self, x):
         x = self.classifier(x)
         return x
 
-class LoicGCN(torch.nn.Module):
-    def __init__(self, edge_index, edge_weight, n_timepoints=50, n_classes=2):
+class ChebConvBlock(torch.nn.Module):
```
Comment (htwangtw) [inline gcn_package/models/gcn.py:51]: Do we want to make this private?

File: gcn_package/models/gcn.py [hunk 2]
```diff
@@ -48,41 +48,132 @@ def forward(self, x):
...
+        # if self.use_edge_pooling:
+        #     batch_vector = torch.from_numpy(
+        #         np.array(range(x.size(0)), dtype=int))
+        #     x, edge_index, _, _ = self.edge_pool(
+        #         x, edge_index=self.edge_index, batch=batch_vector)
+
+        return x
+
+
+class LinearBlock(torch.nn.Module):
```
Comment (htwangtw) [inline gcn_package/models/gcn.py:85]: Same as L51

**Review by htwangtw — APPROVED** [review]
LGTM. The test runs without an issue.
The only part I am nitpicking is do we want to make the "block" classes private and only expose `CustomGCN` to users?

File: gcn_package/models/gcn.py (hunk 1 above)
Comment (ltetrel) [reply gcn_package/models/gcn.py:51]: wdym ? I don't care personally.

Comment (ltetrel) [reply gcn_package/models/gcn.py:51]: Oh sorry okay I see. I don't think python has concept of private classes, how would you do it?

Comment (ltetrel) [reply gcn_package/models/gcn.py:51]: Actually I am doing standalone tests for these blocks, so making them private will complexify testing.
 https://github.com/SIMEXP/gcn_package/blob/6dbd9df9f7a44728d0563789fa519767ddeeda47/gcn_package/tests/test_models.py#L22-L44
